src/Forms/Forms.js

```diff
@@ -227,76 +228,95 @@ FormSelect.propTypes = {
-                        {inputItem.label}
-                    </label>
-                </div>
-            ));
-        }
-        return <div>{result}</div>;
+        const { children, disabled, isInline, onChange, ...props } = this.props;
+
+        return (
+            <div
+                {...props}>
+                {React.Children.map(children, child => {
+                    if (React.isValidElement(child)) {
+                        return React.cloneElement(child, {
+                            disabled: child.props.disabled ? child.props.disabled : disabled,
+                            inline: child.props.inline ? child.props.inline : isInline,
+                            name: child.props.name ? child.props.name : this.groupId,
+                            onChange: child.props.onChange ? child.props.onChange : onChange
+                        });
+                    } else {
+                        return child;
+                    }
+                })}
+            </div>
+        );
     }
 }
-FormRadio.displayName = 'FormRadio';
 
-FormRadio.propTypes = {
-    inputs: PropTypes.array.isRequired,
-    defaultChecked: PropTypes.string,
+FormRadioGroup.propTypes = {
+    children: PropTypes.node,
     disabled: PropTypes.bool,
-    isInline: PropTypes.bool
+    isInline: PropTypes.bool,
+    onChange: PropTypes.func
+};
+
+FormRadioGroup.defaultProps = {
+    onChange: () => {}
 };
 
-FormRadio.propDescriptions = {
-    defaultChecked: 'The `id` of the element selected by default.',
-    inputs: 'Configuration settings for each radio button in the group.',
+FormRadioGroup.propDescriptions = {
     isInline: 'Set to **true** to display radio buttons in a row.'
 };
+
+export const FormRadioItem = ({ checked, children, className, disabled, id, inline, name, value, ...props }) => {
+    const classes = classnames(
+        className,
+        'fd-form__item',
+        'fd-form__item--check',
+        {
+            'fd-form__item--inline': inline
+        }
+    );
+
+    return (
+        <div className={classes} key={id}>
+            <label className='fd-form__label'>
+                <input
+                    {...props}
+                    checked={checked}
+                    className='fd-form__control'
+                    disabled={disabled}
+                    id={id}
+                    name={name}
+                    type='radio'
+                    value={value} />
+                {children}
+            </label>
+        </div>
+    );
+};
+
+FormRadioItem.propTypes = {
+    children: PropTypes.node.isRequired,
+    checked: PropTypes.bool,
+    className: PropTypes.string,
+    defaultChecked: PropTypes.bool,
+    disabled: PropTypes.bool,
+    id: PropTypes.string,
+    inline: PropTypes.bool,
+    name: PropTypes.string,
+    value: PropTypes.string
+};
+
+FormRadioItem.propDescriptions = {
+    checked: 'Set to **true** when radio input is checked and a controlled component.',
```

Comment:
How does it handle the `checked` prop if more than one item is set to true?

Reply:
I would guess React would be the one in charge of that behavior when they receive both? It can't be both a controlled and uncontrolled input at the same time that is for sure.